### PR TITLE
Fixes for compilation errors with the SOF plugin

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

**Review by RanderWang — APPROVED** [review]
LGTM

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@lrudyX @wszypelt not expecting this to fail as it's build fixes (and IIUC would fail on all platforms if broken). Good to merge ?

**Comment by lrudyX** [comment]
> @lrudyX @wszypelt not expecting this to fail as it's build fixes (and IIUC would fail on all platforms if broken). Good to merge ?

It looks like QB problem, I will rerun this.

File: src/include/ipc4/module.h [hunk 1]
```diff
@@ -64,7 +64,7 @@ union ipc4_extended_param_id {
 		uint32_t parameter_type     : 8;
 		uint32_t parameter_instance : 24;
 	} part;
-} __packed __aligned(4);
+} __attribute__((packed, aligned(4)));
```
Comment (btian1) [inline src/include/ipc4/module.h:67]: I noticed kernel use left expression for alignment, and sof fw use right ones, any history on this?

Comment (kv2019i) [reply src/include/ipc4/module.h:67]: "__packed" is a linux kernel shortcut, not necessarily available in generic toolchains (like for SOF plugin).

**Review by kv2019i — APPROVED** [review]
We really need to add a CI test to check the SOF plugin build. Otherwise this will be continuous....

**Pull request merged**